synapse/lib/config.py

```diff
@@ -0,0 +1,42 @@
+import os
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+
+class Config:
+
+    def __init__(self, confdefs):
+        self.conf = {}
+        self.norms = {}
+        self.confdefs = confdefs
+
+        for name, defval, norm in confdefs:
+            self.norms[name] = norm
+            self.conf[name] = defval
+
+    def __iter__(self):
+        return iter(list(self.conf.items()))
+
+    async def loadConfDict(self, conf):
+        for name, valu in conf.items():
+            await self.set(name, valu)
+
+    async def loadConfYaml(self, *path):
+        conf = s_common.yamlload(*path)
+        return await self.loadConfDict(conf)
+
+    async def loadConfEnvs(self, envn):
+        for name, defval, norm in self.confdefs:
+            envv = os.getenv(f'{envn}_{name}'.upper())
+            if envv is not None:
+                await self.set(name, envv)
+
+    def get(self, name):
+        return self.conf.get(name)
+
+    async def set(self, name, valu):
```

Comment:
why is this async?

Reply:
To account for potentially having one of these eventually persist data in a hive or remotely etc.